### ISSUE TITLE
NGINX ingress controller 0.25 does not work on CPUs without SSE4.2 instruction set support

### ISSUE THREAD

**Issue by scndel** [rancher/rancher#23307]
**What kind of request is this (question/bug/enhancement/feature request):**

bug

**Steps to reproduce (least amount of steps as possible):**

setup a fresh rancher 2.3 with a custom cluster on the same machine

**Result:**

nginx-ingress-controller not ready
![Screenshot 2019-10-09 at 16 29 50](https://user-images.githubusercontent.com/17161227/66492641-08417b00-eab5-11e9-9668-eabf032d2a33.png)

````
W1009 14:27:01.767297 6 queue.go:130] requeuing initial-sync, err
-------------------------------------------------------------------------------
Error: signal: illegal instruction (core dumped)
-------------------------------------------------------------------------------
I1009 14:27:01.894015 6 main.go:158] Error during shutdown: signal: illegal instruction (core dumped)
I1009 14:27:01.894125 6 main.go:162] Handled quit, awaiting Pod deletion
E1009 14:27:11.285497 6 checker.go:41] healthcheck error: Get http+unix://nginx-status/healthz: dial unix /tmp/nginx-status-server.sock: connect: no such file or directory
I1009 14:27:11.894291 6 main.go:165] Exiting with 1
````

**Other details that may be helpful:**

retried 3 fresh full install (rancher + cluster), was working fine with v2.2.8

I used the recommended docker port mapping not to mix up with ingress :
```
sudo docker run -d --restart=unless-stopped -p 8080:80 -p 8443:443 -v /etc/rancher/ssl/:/etc/rancher/ssl/ -v /opt/rancher:/var/lib/rancher --name rancher-manager rancher/rancher:v2.3.0
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):

v2.3.0

- Installation option (single install/HA):

single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):

Custom

- Machine type (cloud/VM/metal) and specifications (CPU/memory):

metal Ubuntu 16.04.6 2 cores 4GB

- Kubernetes version (use `kubectl version`):

```
v1.15.4
```

- Docker version (use `docker version`):

```
18.09.9
```

**Comment by scndel** [comment]
Load balancing of traefik from my AWS K3S cluster *imported* associated to the same Rancher works fine so issue is definitely linked with the nginx-ingress-controller of my *custom* docker cluster located on the same metal. 
This can be of course a cause of perfs issue altough basic cluster monitoring doesn't show any pressure :
<img width="1245" alt="Screenshot 2019-10-09 at 22 45 46" src="https://user-images.githubusercontent.com/17161227/66518928-94b86180-eae6-11e9-9b32-50bdfae97238.png">

**Comment by superseb** [comment]
Please post exact specs of the metal machine (see https://github.com/kubernetes/ingress-nginx/issues/4311). When the previous version with v2.2.8 worked perfectly, there must be something in the latest included version.

**Comment by scndel** [comment]
Indeed my poor old metal doesn't support **sse4_2** . After 10 years of good service, it's still good at testing but can we name this a relevant bug ? For Rancher, this lappy is now definetely retired.

````
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 23
model name	: Intel(R) Core(TM)2 Duo CPU     T8300  @ 2.40GHz
stepping	: 6
microcode	: 0x60f
cpu MHz		: 800.000
cache size	: 3072 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 2
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 10
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm sse4_1 lahf_lm kaiser tpr_shadow vnmi flexpriority dtherm ida
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs
bogomips	: 4787.73
clflush size	: 64
cache_alignment	: 64
address sizes	: 36 bits physical, 48 bits virtual
power management:

processor	: 1
vendor_id	: GenuineIntel
cpu family	: 6
model		: 23
model name	: Intel(R) Core(TM)2 Duo CPU     T8300  @ 2.40GHz
stepping	: 6
microcode	: 0x60f
cpu MHz		: 1200.000
cache size	: 3072 KB
physical id	: 0
siblings	: 2
core id		: 1
cpu cores	: 2
apicid		: 1
initial apicid	: 1
fpu		: yes
fpu_exception	: yes
cpuid level	: 10
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm sse4_1 lahf_lm kaiser tpr_shadow vnmi flexpriority dtherm ida
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs
bogomips	: 4787.73
clflush size	: 64
cache_alignment	: 64
address sizes	: 36 bits physical, 48 bits virtual
power management:
```

**Comment by scndel** [comment]
I guess workaround is to stay with Rancher 2.2 or to wait for #14007.
Or @superseb suggestion of #15669 is still valid for Rancher 2.3 ?

**Comment by bjornmagnusson** [comment]
We also encountered this issue when deploying a fresh install of Rancher 2.3.x

**Comment by tryhardest** [comment]
Very imminently time to abandon the nightmare year of Rancher. Ridiculous.

**Comment by vincent99** [comment]
We are just deploying the upstream software.  So we support whatever they do, which apparently no longer includes 10+ year old CPUs.  You can easily disable nginx and find/deploy your own alternative ingress controller that does work with legacy hardware, or k3s supports Traefik which apparently still supports it.

**Comment by onegreyonewhite** [comment]
@vincent99 Maybe it would be much better to upgrade the version of the image to at least 0.27.1 (or 0.28.0)?
This version works great on 10+ year old CPUs.

**Comment by vincent99** [comment]
We will look at and can likely do that as part of regular upgrades of dependencies for 2.4.

**Comment by onegreyonewhite** [comment]
And when is version 2.4 planned with an updated ingress? We are really looking forward to it. How could we help with this?

**Comment by onegreyonewhite** [comment]
At now, we have to change DaemonSet for nginx-ingress-controller after install/update cluster:
```yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  ...
  name: nginx-ingress-controller
  namespace: ingress-nginx
  ...
spec:
  ...
      containers:
      ...
        image: quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.27.1
        ...
        name: nginx-ingress-controller
        securityContext:
          ...
          runAsUser: 101
        ...
```

And it works fine (faster scaling, because the image has become 80mb smaller). Tested on 3 clusters.

**Comment by nuBacuk** [comment]
> At now, we have to change DaemonSet for nginx-ingress-controller after install/update cluster:
> 
> ```yaml
> apiVersion: apps/v1
> kind: DaemonSet
> metadata:
>   ...
>   name: nginx-ingress-controller
>   namespace: ingress-nginx
>   ...
> spec:
>   ...
>       containers:
>       ...
>         image: quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.27.1
>         ...
>         name: nginx-ingress-controller
>         securityContext:
>           ...
>           runAsUser: 101
>         ...
> ```
> 
> And it works fine (faster scaling, because the image has become 80mb smaller). Tested on 3 clusters.

I did not manage to run your solution.
There are no fatal errors, just warnings, but the result is the same.

**Comment by scndel** [comment]
Thanks for your testing @nuBacuk, we can now reuse nginx ! 
Here's an easy fix if you don't want to use nginx (and more generally to remove properly ingress) on a Rancher custom cluster :
https://github.com/rancher/rancher/issues/14074#issuecomment-402663361

**Comment by horihel** [comment]
just FYI as google lists this as a search result: Rancher 2.4 did not fix this as of 2.4.3 although the issue has been removed from the "known issues".
It's still a problem.

**Comment by jwh1981** [comment]
Definitely still an issue! What worked for me though (for a home lab not production) on a Phenom II X3 (which does not have SSE 4.2): I edited the YAML for nginx-ingress-controller and changed nginx-ingress-controller:0.25.1-rancher1 to 0.21.0-rancher3

After a few it updated just fine with 0.21.0-rancher3 and seems to work from my testing. However, this is an older version and there could be unforeseen consequences (like attracting the attention of aliens)
